dev/breeze/src/airflow_breeze/ci/build_image.py

```diff
@@ -0,0 +1,93 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# from airflow_breeze.utils import run_command
+import os
+import shlex
+
+from airflow_breeze.breeze import get_airflow_sources_root
+from airflow_breeze.cache import check_cache_and_write_if_not_cached
+from airflow_breeze.ci.build_params import BuildParams
+from airflow_breeze.utils import run_command
+
+PARAMS_CI_IMAGE = [
+    "python_base_image",
+    "airflow_version",
+    "airflow_branch",
+    "airflow_extras",
+    "airflow_pre_cached_pip_packages",
+    "additional_airflow_extras",
+    "additional_python_deps",
+    "additional_dev_apt_command",
+    "additional_dev_apt_deps",
+    "additional_dev_apt_env",
+    "additional_runtime_apt_command",
+    "additional_runtime_apt_deps",
+    "additional_runtime_apt_env",
+    "upgrade_to_newer_dependencies",
+    "constraints_github_repository",
+    "airflow_constraints_reference",
+    "airflow_constraints",
+    "airflow_image_repository",
+    "airflow_image_date_created",
+    "build_id",
+    "commit_sha",
+]
+
+PARAMS_TO_VERIFY_CI_IMAGE = [
+    "dev_apt_command",
+    "dev_apt_deps",
+    "runtime_apt_command",
+    "runtime_apt_deps",
+]
+
+
+def construct_arguments_docker_command(ci_image: BuildParams) -> str:
+    args_command = ""
+    for param in PARAMS_CI_IMAGE:
+        args_command += " --build-arg " + param.upper() + "=" + str(getattr(ci_image, param))
+    for verify_param in PARAMS_TO_VERIFY_CI_IMAGE:
+        param_value = str(getattr(ci_image, verify_param))
+        if len(param_value) > 0:
+            args_command += " --build-arg " + verify_param.upper() + "=" + param_value
+    args_command += " " + ci_image.docker_cache_ci_directive
+    return args_command
+
+
+def construct_docker_command(ci_image: BuildParams):
+    arguments = construct_arguments_docker_command(ci_image)
+    final_command = (
+        "docker build"
+        + arguments
+        + f" -t {ci_image.airflow_ci_image_name} --target main . -f "
+        + os.path.join(get_airflow_sources_root(), 'Dockerfile.ci')
+    )
+    return final_command
+
+
+def build_image():
+    ci_image_params = BuildParams()
+    is_cached, value = check_cache_and_write_if_not_cached(
+        "PYTHON_MAJOR_MINOR_VERSION", ci_image_params.pythonVersion
+    )
+    if is_cached:
+        ci_image_params.pythonVersion = value
+    # Check params from cache layer and use that for building the image
+    # First pick the params that are in cache and is used for building the ci image like python version
+    cmd = construct_docker_command(ci_image_params)
+    print("Command to run:", cmd)
+    output = run_command(shlex.split(cmd), capture_output=True, text=True)
```

Comment:
We should avoid using shlex.split as this does not always work properly. Whenever possible, we should always prefer to build arrays of arguments from scratch.

Reply:
@mik-laj As the cmd is dynamically built using function i have used shlex.split to break the cmd. Could you suggest some ways to do it in better ways?

Reply:
Agree with @mik-laj . the "construct_docker_command" should return ["docker", "build", .....] array.

Reply:
ok, @potiuk Let me see how I could fix this.